hw/display/virtio-dmabuf.c

```diff
@@ -13,60 +13,76 @@
 #include "hw/virtio/virtio-dmabuf.h"
 
 
-static struct shared_resources {
-    GHashTable *uuids;
-    bool locked;
-} res = {
-    .uuids = NULL,
-    .locked = false,
-};
+#define UUID_TO_POINTER(i) \
+    (gpointer) (g_intern_static_string(qemu_uuid_unparse_strdup((&i))))
 
-bool virtio_add_dmabuf(QemuUUID uuid, int udmabuf_fd)
+
+static GMutex lock;
+static GHashTable *resource_uuids;
+
+
+static bool virtio_add_resource(QemuUUID uuid, gpointer value)
+{
+    if (resource_uuids == NULL) {
+        resource_uuids = g_hash_table_new_full(NULL, NULL, NULL, g_free);
+    }
+    gpointer struuid = UUID_TO_POINTER(uuid);
+    if (g_hash_table_lookup(resource_uuids, struuid) != NULL) {
+        return false;
+    }
+    return g_hash_table_insert(resource_uuids, struuid, value);
+}
+
+static gpointer virtio_lookup_resource(QemuUUID uuid)
 {
-    /*
-     * Check that the fd is valid.
-     */
-    return virtio_add_resource(uuid, GINT_TO_POINTER(udmabuf_fd));
+    if (resource_uuids == NULL) {
+        return (gpointer) NULL;
```

Comment:
Is the casting to `gpointer` required?

Reply:
Not really, I wanted to be explicit on the return type when it only had the function signature. But it could be omitted here.